osu.Game/Screens/Multiplayer/MultiUserPanel.cs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Users;
+
+namespace osu.Game.Screens.Multiplayer
+{
+    public class MultiUserPanel : ClickableContainer, IStateful<MultiUserPanel.UserState>
+    {
+        private UserState state;
+        public UserState State
+        {
+            get { return state; }
+            set
+            {
+                if (state == value)
+                    return;
+
+                state = value;
+                switch (state)
+                {
+                    case UserState.Guest:
+                        statusString = "Waiting in a Room";
+                        break;
+
+                    case UserState.Host:
+                        statusString = "Hosting a Room";
+                        break;
+                }
+                statusSprite.Text = statusString;
+            }
+        }
+
+        private string userName;
+        private string statusString;
+        private OsuSpriteText userSprite;
+        private OsuSpriteText statusSprite;
+
+        public const int PANEL_HEIGHT = 100;
+        public const int STATUS_HEIGHT = 33;
+        public const float PANEL_WIDTH = 0.25f;
+
+
+        public MultiUserPanel(User user = null)
+        {
+            userName = user.Username;
+
+            RelativeSizeAxes = Axes.X;
+            Height = PANEL_HEIGHT;
+            Width = PANEL_WIDTH;
+            Masking = true;
+            CornerRadius = 10;
+            BorderThickness = 0;
+            EdgeEffect = new EdgeEffect
+            {
+                Type = EdgeEffectType.Shadow,
+                Colour = new Color4(0, 0, 0, 40),
+                Radius = 5,
+            };
+            Children = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = new Color4(34,34,34, 255),
```

Comment:
format code manually(the IDE formatter doesn't work in nested object initializer lol)